spikeinterface/toolkit/preprocessing/filter.py

```diff
@@ -166,9 +167,7 @@ def __init__(self, recording, freq=3000, q=30, margin_ms=5.0, dtype=None):
         fn = 0.5 * float(recording.get_sampling_frequency())
         coeff = scipy.signal.iirnotch(freq / fn, q)
 
-        if dtype is None:
-            dtype = recording.get_dtype()
-        dtype = _correct_uint(dtype)
+        dtype = fix_dtype(recording, dtype)
```

Comment:
For the notche filter I think, we should support at all uint and raise an error instead and tell the user to use dtype="float32"